core/src/main/java/org/apache/calcite/plan/volcano/RelSet.java

```diff
@@ -211,6 +219,7 @@ public RelSubset add(RelNode rel) {
     final RelSubset subset = getOrCreateSubset(
         rel.getCluster(), traitSet, rel.isEnforcer());
     subset.add(rel);
+    checkAndUpdateOriginalRel(rel);
```

Comment:
This call seems duplicate with the call in addInternal, as subset.add(rel) will call addInternal?